docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,958 @@
+| `ReadinessGates` (new) | `ReadinessGates`            | `ReadinessGates`                    |
+| other fields...        | other fields...             | other fields...                     |
+
+Notes:
+- Both `MinReadySeconds` and `ReadinessGates` should be treated as other in-place propagated fields (changing this should not trigger rollouts).
+- Similarly to Pod's `ReadinessGates`, also Machine's `ReadinessGates` accept only conditions with positive polarity; 
+  The Cluster API project might revisit this in future to stay aligned with Kubernetes or if there are use cases justifying this change.
+
+#### Machine Print columns
+
+| Current           | To be                         |
+|-------------------|-------------------------------|
+| `NAME`            | `NAME`                        |
+| `CLUSTER`         | `CLUSTER`                     |
+| `NODE NAME`       | `PAUSED` (new) (*)            |
+| `PROVIDER ID`     | `NODE NAME`                   |
+| `PHASE` (deleted) | `PROVIDER ID`                 |
+| `AGE`             | `READY` (new)                 |
+| `VERSION`         | `AVAILABLE` (new)             |
+|                   | `UP TO DATE` (new)            |
+|                   | `AGE`                         |
+|                   | `OS-IMAGE` (new) (*)          |
+|                   | `KERNEL-VERSION` (new) (*)    |
+|                   | `CONTAINER-RUNTIME` (new) (*) |
+
+TODO: figure out if can `INTERNAL-IP` (new) (*),   `EXTERNAL-IP` after `VERSION` / before `OS-IMAGE`?  (similar to Nodes...).
+might be something like `$.status.addresses[?(@.type == 'InternalIP')].address` works, but not sure what happens if there are 0 or more addresses...
+  Stefan +1 if possible
+
+(*) visible only when using `kubectl get -o wide`
+
+### Changes to MachineSet resource
+
+#### MachineSet Status
+
+Following changes are implemented to MachineSet's status:
+
+- Update `ReadyReplicas` counter to use the same semantic Machine's `Ready` (today it is computed a Machines with Node Ready) condition and add missing `UpToDateReplicas`.
+- Remove `FailureReason` and `FailureMessage` to get rid of the confusing concept of terminal failures
+- Transition to new, improved, K8s API conventions aligned conditions
+
+Below you can find the relevant fields in Machine Status v1beta2, after v1beta1 removal (end state);
+Below the Go types, you can find a summary table that also shows how changes will be rolled out according to K8s deprecation rules.
+
+```golang
+type MachineSetStatus struct {
+
+    // The number of ready replicas for this MachineSet. A machine is considered ready when Machine's Ready condition is true.
+    // +optional
+    ReadyReplicas int32 `json:"readyReplicas"`
+	
+	// The number of available replicas for this MachineSet. A machine is considered available when Machine's Available condition is true.
+	// +optional
+	AvailableReplicas int32 `json:"availableReplicas"`
```

Comment:
It is a user choice to set MinReadySeconds and thus introduce a difference between Ready and Available, I assume that depending on the rational behind the user's choice it could make sense to look at one or the another.

Personally, I think that now that we have a clean a consistent definition of readineass and availability across the board, as a project we can decide to intentionally deviate from K8s and show only availability/availability replicas on print columns, thus giving an hint on what the users should look at (Kubernetes instead shows both available and ready replicas columns for ReplicaSets and Deployments, ).

But while I'm open to consider dropping ready replicas from print columns, I defer to a separated issues/proposal the idea of removing the readyReplicas field from status of Cluster API resources, because this implies changes to CAPI controllers which are not in scope of this proposal.

Let's hear other opinions
Please add 👍 if you are in favor ready/ready replicas _from print columns only_, or add 👎 if you are in favor of being consistent with K8s (and show both ready/ready replicas and available/available replicas)

Also related to [final adjustements on print columns](https://github.com/kubernetes-sigs/cluster-api/pull/10897/files#r1693965374)

Reply:
I think I'm fine either way (also depends a bit on how much space we have). Are you asking to drop it entirely or move ready into `-o wide`? Should we just decide this later when we fine-tune the columns?

> and show both ready/ready replicas and available/available replicas

What do you mean with ready/ready and available/available?

Reply:
Resolving because we have agreement on the fact that this requires some fine tuning during/after implementation (which is also well documented in the proposal)
Also tracked in https://github.com/kubernetes-sigs/cluster-api/issues/11105